src/storage/ExpiringResourceLocker.ts

```diff
@@ -0,0 +1,17 @@
+import type { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import type { ExpiringLock } from './ExpiringLock';
+import type { ResourceLocker } from './ResourceLocker';
+
+/**
+ * Allows the locking of resources which is needed for non-atomic {@link ResourceStore}s.
+ * Differs from {@Link ResourceLocker} by adding expiration logic.
+ */
+export interface ExpiringResourceLocker extends ResourceLocker {
```

Comment:
We can probably solve this with generics instead.

So `ExpiringResourceLocker` should simply be an alias of `ResourceLocker<T extends ExpiringLock>`.

Reply:
I'm not quite sure what you mean by this. You mean this?
```TypeScript
export class ExpiringResourceLocker<T extends ExpiringLock> implements ResourceLocker {
```
And keep ResourceLocker as `export interface ResourceLocker {` or modify this interface as well?

Reply:
Yes this would require editing the ResourceLocker interface to be something like `export interface ResourceLocker<T extends Lock = Lock>` (so Lock is used as default value if no type was set). The ExpiringResourceLocker can just be a type alias then: `export type TimedResourceLocker<T extends TimedLock = TimedLock> = ResourceLocker<T>;`.

Reply:
Alright but where should I then implement the expiry functionality if TimedResourceLocker is a type and not a class?
Or are TimedResourceLocker and ExpiringResourceLocker two different things because you now call it TimedResourceLocker and TimedLock instead of ExpiringLock.

Reply:
No that is me not paying attention to which names we used initially. 😄 

Just like we have a `SingleThreadedResourceLocker` implementing the `ResourceLocker` interface, we also need something to then implement the `ExpiringResourcLocker` interface, `WrappedExpiringResourceLocker` for example which takes a `ResourceLocker` as input.